front/src/common/Map/Search/MapSearchStation.tsx

```diff
@@ -54,8 +54,8 @@ const MapSearchStation = ({ updateExtViewport, closeMapSearchPopUp }: MapSearchS
   const orderResults = (results: SearchResultItemOperationalPoint[]) =>
     results.slice().sort((a, b) => a.name.localeCompare(b.name) || a.ch.localeCompare(b.ch));
 
-  const searchByTrigrams = useCallback(async () => {
-    const searchQuery = ['=i', ['trigram'], searchState];
+  const searchByTrigrams = async () => {
```

Comment:
I think you can merge the 2 functions searchByTrigrams and searchByCodesAndNames into a unique function: 
```ts
const searchBy = async (searchByTrigrams: boolean) => {...}
```